VideoApp/VideoApp/Video/Room/Room.swift

```diff
@@ -78,6 +81,18 @@ import TwilioVideo
         room?.disconnect()
     }
     
+    func togglePin(participant new: Participant) {
+        let allParticipants: [Participant] = [localParticipant] + remoteParticipants
+        let old = allParticipants.first(where: { $0.isPinned })
+        
+        if let old = old, old !== new {
+            old.isPinned = false
+        }
+        
+        new.isPinned = !new.isPinned
```

Comment:
Would it be clearer to just say `new.isPinned = true`?  Above you aren't saying `old.isPinned = !old.isPinned`.

```suggestion
        new.isPinned = true
```

Reply:
I think `new` really needs to be a toggle. For example if participant B is pinned and the user taps participant B again to unpin them, it really needs to be set to `false`. Make sense?

Reply:
I guess it is short circuiting some how? I am having trouble grasping the logic portion... If old != new then make sure we unpin the old... If old is equal to new, then that is unpinning new, otherwise it is pinning new?

Reply:
Yes I think your understand of the logic is correct. Mutual exclusive toggle is interesting. :)

Trying to think but not sure there is a clearer way to implement.

Reply:
A small readability improvement...I can use the new Swift `toggle` function:
```swift
new.isPinned.toggle()
```

Reply:
I pushed the `toggle` change.